rubocop.yml

```
@@ -131,3 +131,15 @@ Style/TrailingCommaInArrayLiteral:
 
 Style/TrailingCommaInHashLiteral:
   Enabled: false
+
+RSpec/HookArgument:
+  Enabled: false
+
+RSpec/NotToNot:
+  Enabled: false
+
+RSpec/ExampleLength:
+  Enabled: false
+
+RSpec/MultipleExpectations:
```

This cop caps your number of expects at 1

---

Do we often break this?

---

yes - I break this all the time

Mike DiScala  [1 minute ago]
i often do it to have a test thats `it "constructs the thing properly"` and then asserts a few fields separately

or if i want to call `.first` on an array will do `expect(array.length).to eq(1)` and then write assertions about `array.first`

---

FWIW I generally like to break those up into separate specs, it can make it more obvious what the failure is and can prevent some frustration due to errors hiding other errors.

---

I often do that too, but I don't think this is a global enough convention to enforce. Is there a way to downgrade things to warnings so we can advise against it w/o blocking committing?

---

Not that I know of -- I tried looking into it a little while ago and couldn't figure out a way to do it. Might be worth taking another crack at it at some point.

---

kk - I'm not going to spend time on that right now -- are you okay turning this off for now so we can get this merged in? can follow up later if we figure out how to turn it into a warning

---

Sounds like a plan!